src/workspaces/list/WorkspaceStateCell.tsx

```diff
@@ -0,0 +1,149 @@
+      }}
+      renderers={{}} // needed to make typechecker work, because FirstParagraphMarkdownViewer is not typed
+    >
+      {props.description?.toString() || 'No description added'}
+    </FirstParagraphMarkdownViewer>
+  </div>
+);
+
+interface WorkspaceStateCellProps {
+  workspace: Workspace;
+}
+
+export const WorkspaceStateCell = (props: WorkspaceStateCellProps): ReactNode => {
+  const {
+    workspace,
+    workspace: { attributes, state },
+  } = props.workspace;
+  const description = attributes?.description;
+  const errorMessage = workspace.errorMessage;
+  switch (state) {
+    case 'Deleting':
+      return <WorkspaceDeletingCell />;
+    case 'DeleteFailed':
+      return <WorkspaceFailedCell state={state} errorMessage={errorMessage} />;
+    case 'Deleted':
+      return <WorkspaceDeletedCell />;
+    case 'Cloning':
+      return <WorkspaceCloningCell />;
+    case 'CloningContainer':
+      return <WorkspaceCloningCell />;
+    case 'CloningFailed':
+      return <WorkspaceFailedCell state={state} errorMessage={errorMessage} />;
+    default:
+      return <WorkspaceDescriptionCell description={description} />;
+  }
+};
+
+const WorkspaceDeletingCell = (): ReactNode => (
+  <WorkspaceStatusPill iconShape="syncAlt" rotateIcon color={colors.danger}>
+    Workspace deletion in progress
+  </WorkspaceStatusPill>
+);
+
+const WorkspaceCloningCell = (): ReactNode => (
+  <WorkspaceStatusPill iconShape="syncAlt" rotateIcon color={colors.success}>
+    Workspace cloning in progress
+  </WorkspaceStatusPill>
+);
+
+const WorkspaceStatusPill = (props: {
+  iconShape: IconId;
+  rotateIcon?: boolean;
+  color: (number?) => any;
+  children: ReactNode;
+}) => {
+  return (
+    <div
+      style={{
+        backgroundColor: props.color(0.25),
+        width: 'fit-content',
+        paddingTop: '0.25rem',
+        paddingBottom: '0.25rem',
+        paddingLeft: '1rem',
+        paddingRight: '1rem',
+        fontWeight: 500,
+        textAlign: 'center',
+        borderRadius: '1rem',
+        fontSize: '0.875rem',
+      }}
+    >
+      {icon(props.iconShape, {
+        size: 16,
+        style: {
+          animation: props.rotateIcon ? 'rotation 2s infinite linear' : undefined,
+          color: props.color(),
+          textAlign: 'center',
+        },
+      })}
+      <span style={{ margin: '0.5rem' }}>{props.children}</span>
+    </div>
+  );
+};
+
+interface WorkspaceFailedCellProps {
+  state: 'DeleteFailed' | 'CloningFailed';
+  errorMessage?: string;
+}
+
+const WorkspaceFailedCell = (props: WorkspaceFailedCellProps): ReactNode => {
+  const [showDetails, setShowDetails] = useState<boolean>(false);
+
+  const failureMsg = props.state === 'DeleteFailed' ? 'Error deleting workspace' : 'Workspace clone unsuccessful';
+
+  return (
+    <WorkspaceStatusPill iconShape="warning-standard" color={colors.danger}>
+      {failureMsg}
+      {props.errorMessage ? (
+        // eslint-disable-next-line jsx-a11y/anchor-is-valid
```

Comment:
This also shows up now because of the `tsx` conversion, and happens because of the behind-the-schenes-magic in our components, where the resulting element can be either an anchor or div, depending on the arguments passed.

Reply:
So the user is likely to have many of these "See error details" buttons, can you add an aria-label that includes the workspace name? We do this for the action menu, to make it easier to understand which workspace the menu relates to

<img width="1209" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/484484/fa394565-098f-439e-9ff3-01e4d276d6a7">

Reply:
Thanks for updating!